pkg/s3select/csv/reader.go

```diff
@@ -82,10 +82,11 @@ func (rr *recordReader) Read(p []byte) (n int, err error) {
 
 // Reader - CSV record reader for S3Select.
 type Reader struct {
-	args        *ReaderArgs
-	readCloser  io.ReadCloser
-	csvReader   *csv.Reader
-	columnNames []string
+	args         *ReaderArgs
+	readCloser   io.ReadCloser
+	csvReader    *csv.Reader
+	columnNames  []string
+	nameIndexMap map[string]int64
```

Comment:
This variable is not used anywhere, why do you need it ?

Reply:
It is used at https://github.com/minio/minio/blob/e29a6d74baa876f4117f118c3e84e78fe1e8d6b7/pkg/s3select/csv/reader.go#L120
and to keep a local copy of nameIndexMap, preventing from allocating each time csv.Reader.Read()

Reply:
Understood